fdmi/ut/sd_apply_filter.c

```diff
@@ -185,7 +185,7 @@ static void fdmi_sd_apply_filter_internal(const struct m0_filterc_ops *ops)
 
 	M0_ENTRY();
 
-	g_var_str = strdup("test");
+	g_var_str = m0_strdup("test");
```

Comment:
Is this `g_var_str` freed?

Let me check.

Reply:
Yes, it is copied into a buffer in filterc_send_notif_get_next(), then passed down to fdmi and finally freed within fdmi. I know this, because with alloc-prof enabled, an attempt to m0_free() memory allocated with malloc() or strdup() crashes.

Reply:
> I know this, because with alloc-prof enabled, an attempt to m0_free() memory allocated with malloc() or strdup() crashes.

I wonder - how does it work? Because, according to the following code, m0_strdup is just an alias to strdup:
https://github.com/Seagate/cortx-motr/blob/9456562c6c86e2ea67f79c071005484770c0f8fd/lib/string.h#L43

Reply:
Because m0_free() is not necessarily identical to free(). With alloc-profiler enabled it assumes allocated blocks has a certain header.